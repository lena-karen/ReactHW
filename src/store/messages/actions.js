export const ADD_CHAT = 'ADD_CHAT'
export const ADD_MESSAGE = 'ADD_MESSAGE'
export const DELETE_CHAT = 'DELETE_CHAT'

export const addChat = payload => ({type: ADD_CHAT, payload})

export const addMessage = (message, chat_id) => ({type: ADD_MESSAGE, payload: {message, chat_id}})
    

export const addMessageWithThunk = (message, chat_id) => (dispatch, getState) => {
   
    dispatch(addMessage(message, chat_id))
    if (message.author !== 'BOT') {
        const index = message.id + 1;
        const newMessage = {
            id: index,
            text: "I'm BOT",
            author: 'BOT'
        }

        setTimeout(() => dispatch(addMessage(newMessage, chat_id)), 1500)
    }
}
export const deleteChat = (payload) => ({type: DELETE_CHAT, payload})