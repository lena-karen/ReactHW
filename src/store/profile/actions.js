export const toggleName = (data) => ( {type: 'TOGGLE_NAME', payload: data})