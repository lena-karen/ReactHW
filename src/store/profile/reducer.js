const initialState = {
    name: 'Default Name',
    showName: true
}
export const profileReducer = (state = initialState, action) => {
    const {type} = action
    switch (type) {
        case 'TOGGLE_NAME':
            return {
                ...state,
                showName: !state.showName,
            }
        default:
            return state
    }
}