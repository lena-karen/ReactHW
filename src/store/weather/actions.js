export const DOWNLOAD_WEATHER = 'DOWNLOAD_WEATHER'
export const GET_SUCCESS = 'GET_SUCCESS'
export const GET_FAILURE = 'GET_FAILURE'

export const getWeatherRequest = () => ({type: DOWNLOAD_WEATHER})
export const getWeatherSuccess = (payload) => ({type: GET_SUCCESS, payload})
export const getWeatherFailure = (err) => ({type: GET_FAILURE, payload: err})

export const downloadWeather = () => (dispatch) => {
    dispatch(getWeatherRequest())

    fetch('https://api.open-meteo.com/v1/forecast?latitude=52.52&longitude=13.41&hourly=temperature_2m,relativehumidity_2m,rain&daily=sunrise,sunset&current_weather=true&timezone=Europe%2FBerlin&start_date=2022-12-22&end_date=2022-12-26')
        .then(res => res.json())
        .then(json => 
            dispatch(getWeatherSuccess(json.current_weather)))
        .catch(err => dispatch(getWeatherFailure(true))) 
}