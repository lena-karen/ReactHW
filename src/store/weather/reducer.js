import { DOWNLOAD_WEATHER, 
         GET_SUCCESS, 
         GET_FAILURE } from "./actions"

const initialState = {
    current_weather: 
        { 
            temperature: '', 
            windspeed: '', 
            winddirection: '', 
        },
    error: false,
    loading: false
}

export const weatherReducer = (state = initialState, action) => {
    switch (action.type) {
        case DOWNLOAD_WEATHER: 
            return {
                ...state,
                loading: true
            }

        case GET_SUCCESS:
            return {
                ...state,
                current_weather: action.payload,
                loading: false,
                error: false
            }

        case GET_FAILURE: 
            return {
                ...state,
                loading: false,
                error: action.payload
            }
        default: 
            return state
    }
}