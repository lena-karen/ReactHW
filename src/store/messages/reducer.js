
import { ADD_CHAT, ADD_MESSAGE, DELETE_CHAT } from "./actions"

const initialState = []

export const chatsReducer = (state = initialState, action) => {
    switch (action.type) {
        case ADD_CHAT:
            return [
                ...state,
                action.payload
            ]

        case ADD_MESSAGE:
            console.log('state',state)
            const activeChat = state.find(el => el.id == action.payload.chat_id)
            activeChat.messages.push(action.payload.message)

            const newChatList = state.map(el => {
               if (el.id == action.payload.chat_id){
                   return activeChat
                } else return el;
               })
            return newChatList
        
        case DELETE_CHAT:
            return state.filter(el => el.id != action.payload)
  
        default:
            return state
    }
}