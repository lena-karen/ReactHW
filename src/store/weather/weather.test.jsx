import { weatherReducer } from './reducer'
import { getWeatherRequest, getWeatherSuccess, getWeatherFailure } from './actions'

describe('testing weather request reducer', () => {

    it('returns initial state', () => {
        const expectedState = {
            current_weather: 
            { 
                temperature: '', 
                windspeed: '', 
                winddirection: '', 
            },
            error: false,
            loading: false
        }

        const receivedState = weatherReducer(undefined, getWeatherRequest)
        expect(receivedState).toEqual(expectedState)
    })

    it('state snapshot', () => {
        const state = weatherReducer(undefined, '')
        expect(state).toMatchSnapshot()
    })

    it('return state after GET_SUCCESS action', () => {
        const state = weatherReducer(undefined, getWeatherSuccess)
        expect(state.error).toEqual(false)
        expect(state.loading).toEqual(false)
    })

    it('return state after GET_FAILURE action', () => {
        const state = weatherReducer(undefined, getWeatherFailure)
        expect(state.loading).toEqual(false)
    })

    it('snapshot after GET_SUCCESS action', () => {
        const state = weatherReducer(undefined, getWeatherSuccess)
        expect(state).toMatchSnapshot()
    })
})