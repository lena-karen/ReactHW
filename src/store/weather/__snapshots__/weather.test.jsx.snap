// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing weather request reducer reducer snapshot 1`] = `
Object {
  "current_weather": Object {
    "temperature": "",
    "winddirection": "",
    "windspeed": "",
  },
  "error": false,
  "loading": false,
}
`;

exports[`testing weather request reducer snapshot 1`] = `
Object {
  "current_weather": Object {
    "temperature": "",
    "winddirection": "",
    "windspeed": "",
  },
  "error": false,
  "loading": false,
}
`;

exports[`testing weather request reducer snapshot after GET_SUCCESS action 1`] = `
Object {
  "current_weather": Object {
    "temperature": "",
    "winddirection": "",
    "windspeed": "",
  },
  "error": false,
  "loading": false,
}
`;

exports[`testing weather request reducer state snapshot 1`] = `
Object {
  "current_weather": Object {
    "temperature": "",
    "winddirection": "",
    "windspeed": "",
  },
  "error": false,
  "loading": false,
}
`;
